### PR TITLE
can delete archived

### PR DESCRIPTION
**Pull request by ElijahGeorge**
If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### What's this PR do?
Allows message owner to soft-delete archived messages
#### Screenshots (if appropriate):

### CODE REVIEW

File: server/controllers/message.controller.js [hunk 1]
```diff
@@ -4,11 +4,16 @@ import APIError from '../helpers/APIError';
 
 const Message = db.Message;
 
+const useUnscoped = (url) => {
+  const unscopedRoutes = ['/unarchive/', '/delete/'];
+  return unscopedRoutes.find(route => url.includes(route)) != undefined;
```
Comment (jsachs) [inline server/controllers/message.controller.js]: Use `!==` instead.

**Review by jsachs — CHANGES_REQUESTED** [review]
This looks correct. Can we add tests to make sure that (1) it works for both unarchive and delete, and (2) we can actually delete an archived message?

File: server/controllers/message.controller.js (hunk 1 above)
Comment (ElijahGeorge) [reply server/controllers/message.controller.js]: 👍

Comment (ElijahGeorge) [reply server/controllers/message.controller.js]: Good catch

**Review by jsachs — APPROVED** [review]

**Review by sirctseb — APPROVED** [review]

**Pull request merged**